spec/datadog/tracing/contrib/active_record/async_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'datadog/tracing/contrib/support/spec_helper'
+require 'datadog'
+
+require 'spec/datadog/tracing/contrib/rails/support/deprecation'
+
+require 'active_record'
+
+if PlatformHelpers.jruby?
+  require 'activerecord-jdbc-adapter'
+else
+  require 'pg'
+end
+
+RSpec.describe 'ActiveRecord async query instrumentation' do
+  let(:application_record) do
+    stub_const('ApplicationRecord', Class.new(ActiveRecord::Base)).tap do |klass|
+      klass.abstract_class = true
+    end
+  end
+  let!(:widget_class) { stub_const('Widget', Class.new(application_record)) }
+
+  before do
+    skip 'Test applies only to Active Record 7.0 or higher' if ActiveRecord.version < '7.0.0'
+
+    # Reset options (that might linger from other tests)
+    Datadog.configuration.tracing[:active_record].reset!
+
+    Datadog.configure do |c|
+      c.tracing.instrument :concurrent_ruby
```

Comment:
Strictly speaking, this isn't actually necessary, but similar to the weirdness I raised in https://github.com/DataDog/dd-trace-rb/pull/4543#discussion_r2019081549, the interaction between Active Record and Concurrent Ruby is a little odd.